### PR TITLE
add new properties to imageInfo node

### PR DESCRIPTION
**Pull request by acosferreira**
# Description

Add image package's diff when there are an update available to a group. Add image info as distribution, version, date . 

Fixes # ([issue](https://issues.redhat.com/browse/THEEDGE-2009))

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] I run `go fmt ./...` to check that my code is properly formatted
- [X] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

File: pkg/services/devicegroups.go [hunk 1]
```diff
@@ -186,15 +188,36 @@ func GetDeviceImageInfo(images map[int]models.DeviceImageInfo, account string) e
 			latestImage := &deviceImageSet.Images[len(deviceImageSet.Images)-1]
 			latestImageID := latestImage.ID
 
-			if int(latestImageID) > imageID {
+			if int(latestImageID) > imageID && latestImage.CommitID > 0 {
+
 				updAvailable = true
 				CommitID = deviceImageSet.Images[len(deviceImageSet.Images)-1].CommitID
+
+				db.DB.First(&deviceImage.Commit, deviceImage.CommitID)
```
Comment (thom-at-redhat) [inline pkg/services/devicegroups.go:196]: What about errors here?

File: pkg/services/devicegroups.go [hunk 2]
```diff
@@ -186,15 +188,36 @@ func GetDeviceImageInfo(images map[int]models.DeviceImageInfo, account string) e
...
+				if err := db.DB.Model(&deviceImage.Commit).Association("InstalledPackages").Find(&deviceImage.Commit.InstalledPackages); err != nil {
+					s.log.WithField("error", err.Error()).Error("Error when getting InstalledPackafes to CurrentImage")
```
Comment (thom-at-redhat) [inline pkg/services/devicegroups.go:198]: Typo 1

File: pkg/services/devicegroups.go [hunk 3]
```diff
@@ -186,15 +188,36 @@ func GetDeviceImageInfo(images map[int]models.DeviceImageInfo, account string) e
...
+					return err
+				}
+
+				db.DB.First(&latestImage.Commit, latestImage.CommitID)
+				if err := db.DB.Model(&latestImage.Commit).Association("InstalledPackages").Find(&latestImage.Commit.InstalledPackages); err != nil {
+					s.log.WithField("error", err.Error()).Error("Error when getting InstalledPackafes to LatestImage")
```
Comment (thom-at-redhat) [inline pkg/services/devicegroups.go:204]: Typo 2

File: pkg/services/devicegroups.go [hunk 4]
```diff
@@ -186,15 +188,36 @@ func GetDeviceImageInfo(images map[int]models.DeviceImageInfo, account string) e
...
 				updAvailable = true
 				CommitID = deviceImageSet.Images[len(deviceImageSet.Images)-1].CommitID
+
+				db.DB.First(&deviceImage.Commit, deviceImage.CommitID)
+				if err := db.DB.Model(&deviceImage.Commit).Association("InstalledPackages").Find(&deviceImage.Commit.InstalledPackages); err != nil {
+					s.log.WithField("error", err.Error()).Error("Error when getting InstalledPackafes to CurrentImage")
+					return err
+				}
+
+				db.DB.First(&latestImage.Commit, latestImage.CommitID)
```
Comment (thom-at-redhat) [inline pkg/services/devicegroups.go:202]: or errors here?

**Review by thom-at-redhat — CHANGES_REQUESTED** [review]

File: pkg/models/devicegroups.go [hunk 5]
```diff
@@ -27,6 +27,10 @@ type DeviceGroupListDetail struct {
 //DeviceImageInfo is a record of group with the current images running on the device
 type DeviceImageInfo struct {
 	Name            string
+	Vesrion         int
```
Comment (loadtheaccumulator) [inline pkg/models/devicegroups.go:30]: Version typo

**Review by loadtheaccumulator — CHANGES_REQUESTED** [review]

File: pkg/models/devicegroups.go (hunk 5 above)
Comment (acosferreira) [reply pkg/models/devicegroups.go]: thanks!

File: pkg/services/devicegroups.go [hunk 6]
```diff
@@ -186,15 +188,44 @@ func GetDeviceImageInfo(images map[int]models.DeviceImageInfo, account string) e
...
+					}
+				}
+
+				imagePackageDiff = GetDiffOnUpdate(deviceImage, *latestImage)
+
 			}
 
 			images[imageID] = models.DeviceImageInfo{
 				Name:            deviceImage.Name,
+				Vesrion:         deviceImage.Version,
```
Comment (thom-at-redhat) [inline pkg/services/devicegroups.go]: typo is also here

File: pkg/services/devicegroups.go (hunk 3 above)
Comment (acosferreira) [reply pkg/services/devicegroups.go]: Thanks

File: pkg/services/devicegroups.go (hunk 2 above)
Comment (acosferreira) [reply pkg/services/devicegroups.go]: Thanks

**Review by thom-at-redhat — CHANGES_REQUESTED** [review]

**Review by thom-at-redhat — APPROVED** [review]

**Review by loadtheaccumulator — APPROVED** [review]

**Pull request merged**